lib/ip.go

```diff
@@ -18,6 +20,29 @@ func NewIP(ip uint32) IP {
 	return IP(ip)
 }
 
+// RandIP4 returns a new randomly generated IPv4 address.
+func RandIP4() net.IP {
+	ip := make([]byte, 4)
```

Comment:
We can optimize this a bit so that we create a stack-level array rather than use `make` which would go on the heap (although, the Go compiler may already optimize it this way because this `ip` array does not escape from the function).

So:

```go
	ip := [4]byte{0, 0, 0, 0}
	binary.BigEndian.PutUint32(ip, rand.Uint32())
	return net.IPv4(ip[0], ip[1], ip[2], ip[3])
```